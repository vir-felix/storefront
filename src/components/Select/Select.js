import React, { Component } from "react";
import PropTypes from "prop-types";
import MuiSelect from "material-ui/Select";
import MenuItem from "material-ui/Menu/MenuItem";

class Select extends Component {
  static propTypes = {
    inputProps: PropTypes.object,
    onChange: PropTypes.func.isRequired,
    options: PropTypes.arrayOf(PropTypes.object).isRequired,
    value: PropTypes.number
  }

  renderOptions() {
    const { options } = this.props;

    return (
      options.map((option) => (
        <MenuItem key={option.value} value={option.value}>{option.name}</MenuItem>
      ))
    );
  }

  handleChange = (event) => {
    this.props.onChange(event);
  }

  render() {
    const { value, inputProps } = this.props;

    return (
      <MuiSelect
        value={value}
        onChange={this.handleChange}
        inputProps={{ ...inputProps }}
      >
        {this.renderOptions()}
      </MuiSelect>
    );
  }
}

export default Select;
