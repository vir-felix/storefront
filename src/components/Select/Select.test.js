import React from "react";
import renderer from "react-test-renderer";
import Select from "./Select";

const PAGE_SIZES = [
  {
    name: "20 Products",
    value: 20
  },
  {
    name: "60 Products",
    value: 60
  },
  {
    name: "100 Products",
    value: 100
  }
];

test("basic snapshot", () => {
  const component = renderer.create((
    <Select
      value={20}
      options={PAGE_SIZES}
      inputProps={{
        name: "pageSize",
        id: "page-size"
      }}
      onChange={ () => true }
    />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
