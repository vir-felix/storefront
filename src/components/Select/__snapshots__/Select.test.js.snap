// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  className="MuiInput-root-10"
>
  <div
    className="MuiSelect-root-5"
  >
    <div
      aria-haspopup="true"
      aria-owns={null}
      aria-pressed="false"
      className="MuiSelect-select-6 MuiSelect-selectMenu-7 Select-selectMenu-3 MuiInput-input-18"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      20 Products
    </div>
    <input
      aria-invalid={undefined}
      aria-required={undefined}
      autoComplete={undefined}
      autoFocus={undefined}
      defaultValue={undefined}
      id="page-size"
      name="pageSize"
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder={undefined}
      readOnly={undefined}
      required={undefined}
      rows={undefined}
      type="hidden"
      value={20}
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-23 MuiSelect-icon-9"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
